c++/samples/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+cmake_minimum_required (VERSION 2.8.12)
+
+project ("Cap'n'proto sample" CXX)
+
+# Needed to include generated Cap'n'proto headers
+set (CMAKE_INCLUDE_CURRENT_DIR ON)
+
+# Include CMake scripts
+include (FindPkgConfig)
+set (CMAKE_MODULE_PATH
+    "${PROJECT_SOURCE_DIR}/../cmake"
+    "${PROJECT_SOURCE_DIR}/cmake"
+)
+
+# Enable tests
+include(CTest)
+
+# Check for C++11 support
+include (C++11)
+CheckCXX11(FATAL_ERROR)
+
+# Add "-Wall" to compiler flags
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+
+# Find Cap'n'proto
+find_package (Capnp REQUIRED)
+add_definitions(${CAPNP_DEFINITIONS})
+
+# Compile .capnp files
+capnp_generate_cpp(CAPNPSRC_addressbook CAPNPHDR_addressbook "${PROJECT_SOURCE_DIR}/addressbook.capnp")
+capnp_generate_cpp(CAPNPSRC_calculator CAPNPHDR_calculator "${PROJECT_SOURCE_DIR}/calculator.capnp")
+
+# Includes
+include_directories (
+    "${PROJECT_SOURCE_DIR}"
+    ${CAPNP_INCLUDE_DIRS}
+)
+
+# Targets
+add_executable (addressbook
+    addressbook.c++
+    "${CAPNPSRC_addressbook}"
+    "${CAPNPHDR_addressbook}"
+)
+target_link_libraries (addressbook ${CAPNP_LIBRARIES})
+
+add_executable (calculator-client
+    calculator-client.c++
+    "${CAPNPSRC_calculator}"
+    "${CAPNPHDR_calculator}"
+)
+target_link_libraries (calculator-client ${CAPNP_LIBRARIES})
+
+add_executable (calculator-server
+    calculator-server.c++
+    "${CAPNPSRC_calculator}"
+)
+target_link_libraries (calculator-server ${CAPNP_LIBRARIES})
+
+# Tests
+enable_testing()
+
+set (ADDRESSBOOK_EXEC "${PROJECT_BINARY_DIR}/addressbook")
```

Comment:
Could use a generator here instead: `$<TARGET_FILE:addressbook>` (and similarly for the calculator).